### DISCUSSION TITLE
Downloading individual packages

### DISCUSSION THREAD

**Discussion by Lus0rius** [Zdendaki/RailworksDownloader#28] [Ideas]
Hi ! Thanks for this software, I really like it !

I think it would be a great improvement if we could download individual packages (for example through a button in the Route dependencies window) or even (if possible) individual files when clicking on a package in the software.

Greetings from Switzerland !

**Comment by JachyHm** [comment]
Hi!
We appreciate your interest and it's nice to see that this project is slowly spreading out of Czechia too.

Downloading of individual packages is currently only possible from [web-side (dls.rw.jachyhm.cz)](https://dls.rw.jachyhm.cz).
Althought we planned to add support for downloading packages directly from app (actually the window is still present in last release), it was considered "double implementation" as we already had this, yet only slightly different way.

Downloading of individual files would mean redoing whole project's logic and it would not come with such benefits, so that it would pay off for the ammount of work.
Also you usually need other files like meshes or textures too, so it wouldn't make sense to download only single file.
This should be though controlled by package author, that one package includes only files necesarry for it to work and not other files that could posibly be separated into another package.

I hope this helped to clarify your questions.
J :)

**Comment by Lus0rius** [comment]
Thanks for your reply !
I know that downloading individual packages is already possible through the website, I just thought it would be more convenient to do it directly in-app.
It's ok for individual files, I already suspected the implementation to be heavy.

Have a great evening !

**Reply by JachyHm** [reply]
There are basically two ways of doing it.
One will significantly lower download speed, the other one will either increase diskspace needed more than a hundred times, or use so much compute power and thus almost inevitably increasing download time.
Therefore I don't see any benefits which would pay off for doing it such way.
